Add DescopeException for structured errors (#24)
descope/descope-flutter: README.md

- [Quickstart](#quickstart)
- [Session Management](#session-management)
- [Custom Claims](#custom-claims)

> - [Error handling via DescopeException](#error-handling)

Reviewer: ```suggestion
- [Error handling](#error-handling)
```

Note that any custom claims added via this method are considered insecure and will
be nested under the `nsec` custom claim.

## Error Handling

> All operations in the Descope SDK will throw a `DescopeException` in the event an error occurs.
> There are several ways to catch and handle a `DescopeException` thrown by a Descope SDK
> operation, and you can use whichever one is more appropriate in each specific use case.

Reviewer: ```suggestion
All authentication operations throw a `DescopeException` in case of a failure. There are several
ways to catch and handle a `DescopeException`, and you can use whichever one is more
appropriate in each specific use case.
```

final authResponse = await Descope.password.signUp(loginId: 'desmond_c@mail.com', password: 'cleartext-password',
    details: SignUpDetails(name: 'Desmond Copeland'));
```
## Error Handling via `DescopeException`

> All of the authentication operations will throw a `DescopeException` in the event an error
> is encountered. There are several ways to catch and handle a `DescopeException` thrown by a Descope SDK
> operation, and you can use whichever one is more appropriate in each specific use case.

Reviewer: ```suggestion
All authentication operations throw a `DescopeException` in case of a failure. There are several
ways to catch and handle a `DescopeException` thrown by a Descope SDK operation, and you
can use whichever one is more appropriate in each specific use case.
```
